pkg/credentials/credentials.go

```diff
@@ -254,6 +254,12 @@ func AppendCloudProviderVolumes(dpa *oadpv1alpha1.DataProtectionApplication, ds
 
 // add plugin specific specs to velero deployment
 func AppendPluginSpecificSpecs(dpa *oadpv1alpha1.DataProtectionApplication, veleroDeployment *appsv1.Deployment, veleroContainer *corev1.Container, providerNeedsDefaultCreds map[string]bool, hasCloudStorage bool) error {
+
+	init_container_resources := corev1.ResourceRequirements{}
+	if len(veleroDeployment.Spec.Template.Spec.Containers) > 0 {
```

Comment:
You are trying to grab the container resources assigned to velero container, right ? This code needs a container name check, currently this snippet just grabs the first container that it encounters in velero deployment. The check will make it future proof and free of inaccuracies.